src/ResourceManager/Network/Commands.Network/DdosProtectionPlan/GetAzureRMDdosProtectionPlanCommand.cs

```diff
@@ -0,0 +1,88 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AutoMapper;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using CNM = Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Network.Automation
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmDdosProtectionPlan"), OutputType(typeof(PSDdosProtectionPlan))]
+    public partial class GetAzureRmDdosProtectionPlan : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Specifies the name of the DDoS protection plan resource group.",
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("ResourceName")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Specifies the name of the DDoS protection plan.",
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if(!string.IsNullOrEmpty(this.Name))
+            {
+                var vDdosProtectionPlan = this.NetworkClient.NetworkManagementClient.DdosProtectionPlans.Get(ResourceGroupName, Name);
```

Comment:
@murilogr does this `Get` method handle when `ResourceGroupName` is `null`? If not, we should have two parameter sets to prevent the user from providing `Name` without `ResourceGroupName`:
- One set with an optional `ResourceGroupName` parameter
- One set with mandatory `ResourceGroupName` and `Name` parameters

Reply:
I tried it and it validates it:

```
PS D:\azure-powershell> Get-AzureRmDdosProtectionPlan -Name  DdosProtectionPlanName
Get-AzureRmDdosProtectionPlan : 'resourceGroupName' cannot be null.
At line:1 char:1
+ Get-AzureRmDdosProtectionPlan -Name  DdosProtectionPlanName
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzureRmDdosProtectionPlan], ValidationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Network.Automation.GetAzureRmDdosProtectionPlan
```

Reply:
@murilogr the idea is that we don't have a scenario that leads users to get this error ☺️ if we have the split parameter sets where the user provides both `ResourceGroupName` and `Name` _or_ an optional `ResourceGroupName`, then there is no situation where they'd get this error because they'd never be able to provide just the `Name` parameter

Reply:
Agreed :) Added parameter sets